frontend/viewer/src/lib/components/ui/switch/switch.svelte

```diff
@@ -0,0 +1,27 @@
+<script lang="ts">
+  import {cn} from '$lib/utils.js';
+  import {Switch as SwitchPrimitive, type WithoutChildrenOrChild} from 'bits-ui';
+
+  let {
+    ref = $bindable(null),
+    class: className,
+    checked = $bindable(false),
+    ...restProps
+  }: WithoutChildrenOrChild<SwitchPrimitive.RootProps> = $props();
+</script>
+
+<SwitchPrimitive.Root
```

Comment:
I'd like to see support for an icon, I find that usually it's pretty unclear what's considered off or on.
Material guidlines: https://m3.material.io/components/switch/guidelines#7e33ceea-6cdf-4542-8d50-4b75ca23d887
They also suggest making the Thumb icon smaller when checked, so we might want to do that too

Reply:
> The "on" state of the switch is indicated by a larger handle size

Is that what you meant? It's seems the handle is smaller when the switch is turned off. Then when you activate it, it expands to fill the entire track and then shrinks slightly as it transitions to the right.

Reply:
yes that's what I meant lol